docs/reference/modules/discovery/quorums.asciidoc

```diff
@@ -0,0 +1,181 @@
+[[modules-discovery-quorums]]
+=== Quorum-based decision making
+
+Electing a master node and changing the cluster state are the two fundamental
+tasks that master-eligible nodes must work together to perform. It is important
+that these activities work robustly even if some nodes have failed, and
+Elasticsearch achieves this robustness by only considering each action to have
+succeeded on receipt of responses from a _quorum_, a subset of the
+master-eligible nodes in the cluster. The advantage of requiring only a subset
+of the nodes to respond is that it allows for some of the nodes to fail without
+preventing the cluster from making progress, and the quorums are carefully
+chosen so as not to allow the cluster to "split brain", i.e. to be partitioned
+into two pieces each of which may make decisions that are inconsistent with
+those of the other piece.
+
+Elasticsearch allows you to add and remove master-eligible nodes to a running
+cluster. In many cases you can do this simply by starting or stopping the nodes
+as required, as described in more detail in the
+<<modules-discovery-adding-removing-nodes,section on adding and removing
+nodes>>.
+
+As nodes are added or removed Elasticsearch maintains an optimal level of fault
+tolerance by updating the cluster's _voting configuration_, which is the set of
+master-eligible nodes whose responses are counted when making decisions such as
+electing a new master or committing a new cluster state. A decision is only made
+once more than half of the nodes in the voting configuration have responded.
+Usually the voting configuration is the same as the set of all the
+master-eligible nodes that are currently in the cluster, but there are some
+situations in which they may be different.
+
+To be sure that the cluster remains available you **must not stop half or more
+of the nodes in the voting configuration at the same time**. As long as more
+than half of the voting nodes are available the cluster can still work normally.
+This means that if there are three or four master-eligible nodes then the
+cluster can tolerate one of them being unavailable; if there are two or fewer
+master-eligible nodes then they must all remain available.
+
+After a node has joined or left the cluster the elected master must issue a
+cluster-state update that adjusts the voting configuration to match, and this
+can take a short time to complete. It is important to wait for this adjustment
+to complete before removing more nodes from the cluster.
+
+[float]
+==== Setting the initial quorum
+
+When a brand-new cluster starts up for the first time, one of the tasks it must
+perform is to elect its first master node, for which it needs to know the set
+of master-eligible nodes whose votes should count in this first election. This
+initial voting configuration is known as the _bootstrap configuration_ and is
+set in the <<modules-discovery-bootstrap-cluster,cluster bootstrapping
+process>>.
+
```

Comment:
If I'm understanding correctly, it seems like this "Setting the initial quorum" information should be merged with the "Bootstraping a cluster" page. I think this information here is a better overview than what correctly exists in the "Bootstrapping..." page.